packages/shared/src/components/PostOptionsReadingHistoryMenu.tsx

```diff
@@ -45,8 +49,8 @@ const updateReadingHistoryPost =
   ): ((args: { id: string }) => Promise<() => void>) =>
   async () => {
     const history = client.getQueryData<ReadHistoryInfiniteData>(queryKey);
-    const { node } = history.pages[page].readHistory.edges[edge];
-    const updated = { ...node, post: { ...node.post, ...post } };
+    const { node } = history?.pages[page].readHistory.edges[edge] || {};
+    const updated = { ...node, post: { ...node?.post, ...post } };
```

Comment:
The main fix is using the correct query key as seen below, but still making sure this does not break if for some reason history ends up `undefined` is a good idea IMO